cmds/msr/msr.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2012-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// For now, this only works on linux.
+// +build linux
+
+// msr lets you read and write an msr for one or more cores.
+// The cores are specified via a filepath.Glob string.
+// Usage:
+// msr r glob 32-bit-msr-number
+// msr w glob 32-bit-msr-number 64-bit-value
+// For each MSR operation msr will print an error if any.
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+)
+
+const usage = `msr r glob register
+msr w glob register value`
+
+func main() {
+	var err error
+	a := os.Args[1:]
+
+	if len(a) < 3 {
+		log.Fatal(usage)
+	}
+
+	m := msrList(a[1])
+
+	if len(m) == 0 {
+		log.Fatalf("No msrs match '%v'", a[1])
```

Comment:
For some reason, a lot of systems (like Ubuntu) don't auto-load the msr module, so you have to run "modprobe msr" beforehand. In the error message, you might want to including "Make sure Linux was built with CONFIG_X86_MSR=y or the msr module has been loaded".